zerver/tests/test_user_groups.py

```diff
@@ -168,6 +175,25 @@ def test_subgroups_of_role_based_system_groups(self) -> None:
             ],
         )
 
+    def test_is_user_in_group(self) -> None:
+        realm = get_realm("zulip")
+        shiva = self.example_user("shiva")
+        iago = self.example_user("iago")
+        hamlet = self.example_user("hamlet")
+
+        moderators_group = UserGroup.objects.get(
+            name="@role:moderators", realm=realm, is_system_group=True
+        )
+
+        self.assertTrue(is_user_in_group(moderators_group, shiva))
+
+        # Iago is member of a subgroup of moderators group.
+        self.assertTrue(is_user_in_group(moderators_group, iago))
+        self.assertFalse(is_user_in_group(moderators_group, iago, direct_member_only=True))
+
+        self.assertFalse(is_user_in_group(moderators_group, hamlet))
+        self.assertFalse(is_user_in_group(moderators_group, hamlet, direct_member_only=True))
```

Comment:
I extended this test to have a case where `is_user_in_group(moderators_group, hamlet, direct_member_only=True)` is True.